checks.d/network.py

```diff
@@ -440,3 +446,117 @@ def _parse_solaris_netstat(self, netstat_output):
+                proc.close()
+            return int(lines)
+        except IOError:
+            self.log.debug("Unable to read %s." % proc_name)
+
+    @staticmethod
+    def _linux_sockstat_get_size_of_page():
+        return long(check_output(["/usr/bin/getconf", "PAGESIZE"]))
+
+    @staticmethod
+    def _linux_sockstat_get_max_memory(self):
+        sizeOfPageInBytes = Network._linux_sockstat_get_size_of_page()
+        tcpMemFilePath = '/proc/sys/net/ipv4/tcp_mem'
+        try:
+            proc = open(tcpMemFilePath,'r')
+            try:
+                tcpmem = proc.readlines()
+            finally:
+                proc.close()
+
+        except IOError:
+            self.log.debug("Unable to read %s." % tcpMemFilePath)
+        tcpmemline = tcpmem[0].split()
+        maxMemoryInBytes = sizeOfPageInBytes * tcpmemline[2]
+        return long(maxMemoryInBytes)
+
+    @staticmethod
+    def _linux_sockstat(self):
+        proc_name = '/proc/net/sockstat'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.read()
+            finally:
+                proc.close()
+            reg = re.compile("sockets: used (?P<sockets_used>\d+)\n"
+                        "TCP: inuse (?P<tcp_inuse>\d+) orphan (?P<orphans>\d+)"
+                        " tw (?P<tw_count>\d+) alloc (?P<tcp_sockets>\d+)"
+                        " mem (?P<tcp_pages>\d+)\n"
+                        "UDP: inuse (?P<udp_inuse>\d+)"
+                        "(?: mem (?P<udp_pages>\d+))?\n"
+                        "(?:UDPLITE: inuse (?P<udplite_inuse>\d+)\n)?"
+                        "RAW: inuse (?P<raw_inuse>\d+)\n"
+                        "FRAG: inuse (?P<ip_frag_nqueues>\d+)"
+                        " memory (?P<ip_frag_mem>\d+)\n")
+            match = re.match(reg, lines)
+            if not match:
+                self.log.debug("Unable to parse %s." % proc_name)
+                return
+
+            values = []
+            values.append(match.group("sockets_used"))
+            values.append(match.group("tcp_inuse"))
+            values.append(match.group("orphans"))
+            values.append(match.group("tw_count"))
+            values.append(match.group("tcp_sockets"))
+            values.append(match.group("tcp_pages"))
+            values.append(match.group("udp_inuse"))
+            values.append(match.group("udp_pages"))
+            values.append(match.group("udplite_inuse"))
+            values.append(match.group("raw_inuse"))
+            values.append(match.group("ip_frag_nqueues"))
+            values.append(match.group("ip_frag_mem"))
+            metric_names = ['sockstat.sockets.used','sockstat.sockets.tcp.inuse','sockstat.sockets.tcp.orphan','sockstat.sockets.tcp.tw ','sockstat.sockets.tcp.alloc','sockstat.sockets.tcp.mem','sockstat.sockets.udp.inuse','sockstat.sockets.udp.mem','sockstat.sockets.udp.lite','sockstat.sockets.raw.inuse','sockstat.sockets.frag.inuse','sockstat.sockets.frag.memory']
```

Comment:
Instead of defining a list of metric names here, try defining something up top by the other metric list names (above the class definition.

Something like:
```
LINUX_SOCKSTAT_METRICS = [
    ("sockets_used", 'sockstat.sockets.used'),
    ("tcp_inuse", 'sockstat.sockets.tcp.inuse'),
    ("orphans", 'sockstat.sockets.tcp.orphan'),
    ("tw_count", 'sockstat.sockets.tcp.tw'),
    ("tcp_sockets", 'sockstat.sockets.tcp.alloc'),
    ("tcp_pages", 'sockstat.sockets.tcp.mem'),
    ("udp_inuse", 'sockstat.sockets.udp.inuse'),
    ("udp_pages", 'sockstat.sockets.udp.mem'),
    ("udplite_inuse", 'sockstat.sockets.udp.lite'),
    ("raw_inuse", 'sockstat.sockets.raw.inuse'),
    ("ip_frag_nqueues", 'sockstat.sockets.frag.inuse'),
    ("ip_frag_mem", 'sockstat.sockets.frag.memory')
]
```